Core/clim-core/frames.lisp

```diff
@@ -1727,24 +1729,21 @@ have a `pointer-documentation-pane' as pointer documentation,
 
 (defun document-drag-n-drop
     (translator presentation context-type frame event window x y)
-  (when *pointer-documentation-output*
-    (let ((s *pointer-documentation-output*))
-      (window-clear s)
-      (with-end-of-page-action (s :allow)
-	(with-end-of-line-action (s :allow)
-	  (funcall (pointer-documentation translator)
-	   *dragged-object*
-	   :presentation *dragged-presentation*
-	   :destination-object (and presentation
-				    (presentation-object presentation))
-	   :destination-presentation presentation
-	   :context-type context-type
-	   :frame frame
-	   :event event
-	   :window window
-	   :x x
-	   :y y
-	   :stream s))))))
-
-
-
+  (let ((stream *pointer-documentation-output*)
+        (function (pointer-documentation translator)))
```

Comment:
I think it wasn't looking like that in the previous version. Are you sure that `*pointer-documentation-output*` is always bound to a stream and this `when` was unnecessary? Is it certain, that `(pointer-documentation translator)` will return a function?

Everything else looks fine to me, although I'm a bit skeptical about using multiple-value calls.

Reply:
You are right with respect to `*pointer-documentation-output*`, it can be `nil`. I put the guard back in and tested with programs with and without pointer documentation pane. So that should be fine.

`(pointer-documentation translator)` should never be `nil`. I read the spec that way, the `define-drag-and-drop-translator` macro inserts a default value if necessary and the other call to `(pointer-documentation translator)` in the code base also assumes that it is always a function.

mv calls avoid having three helper functions that all check whether `last-event` has already been set (I found that problem while testing the other changes). I it's fine with you, I would keep them.